shared/extension/Configuration.hpp

```diff
@@ -223,27 +223,31 @@ namespace extension {
 // NUClear configuration extension
 namespace NUClear::dsl {
     namespace operation {
+        namespace {
+            [[nodiscard]] std::string getFirstCommandLineArg() {
+                std::shared_ptr<const message::CommandLineArguments> args =
+                    store::DataStore<message::CommandLineArguments>::get();
+                std::vector<char> data(args->at(0).cbegin(), args->at(0).cend());
+                data.push_back('\0');
+                return std::string(data.data());
+            }
+        }  // namespace
```

Comment:
Isn't anon namespace in hpp bad?

Reply:
It makes and compiles a separate version of it for every c++ file that includes this. So for Configuration.hpp that almost everything includes it's probably not great :P

Also, this is clearly not a templated or inline function so what's it even doing in the header in the first place!

Reply:
[it's bad](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file#:~:text=Do%20not%20define%20an%20unnamed%20namespace%20in%20a%20header%20file&text=When%20an%20unnamed%20namespace%20is,used%20within%20that%20translation%20unit.). I forgot lol. there's a [clang-tidy check](https://clang.llvm.org/extra/clang-tidy/checks/google-build-namespaces.html) for it.